### DISCUSSION TITLE
docs/hashmap-array/valid-parentheses/

### DISCUSSION THREAD

**Discussion by giscus** [zrcoder/leetcode#55] [General]
# docs/hashmap-array/valid-parentheses/

从较简单的问题开始：
20. 有效的括号(简单) 给定一个只包括 '('，')'，'{'，'}'，'['，']' 的字符串 s ，判断字符串是否有效。
有效字符串需满足：
左括号必须用相同类型的右括号闭合。 左括号必须以正确的顺序闭合。 每个右括号都有一个对应的相同类型的左括号。 示例 1：
输入： s = “()” 输出： true
示例 2：
输入： s = “()[]{}” 输出： true
示例 3：
输入： s = “(]” 输出： false
提示：
1 <= s.length <= 104 s 仅由括号 '()[]{}' 组成 分析 从前向后遍历，如果遇到一个右括号，其前一个位置一定要是一个匹配的左括号，这样整个字符串才可能有效。
func isValid(s string) bool { stk := make([]rune, 0, len(s)) // 用这个栈维护临时的左括号 pair := map[rune]rune { ')': '(', ']': '[', '}': '{', } for _, ch := range s { switch ch { case ')', ']', '}': // 预期栈顶是匹配的左括号 if len(stk) == 0 || stk[len(stk)-1] !

https://rdcode.netlify.app/docs/hashmap-array/valid-parentheses/

**Comment by zrcoder** [comment]
比较烧脑，需要好好玩玩。
